packages/remix-server-runtime/server.ts

```diff
@@ -74,24 +75,18 @@ export const createRequestHandler: CreateRequestHandlerFunction = (
           createStaticHandlerDataRoutes(build.routes, loadContext)
         );
 
-        let [response, remixRouterResponse] = await Promise.all([
```

Comment:
Let's add an integration test here to post data to a resource route action to prove the bug, and then we can go back to the `Promise.all`.

Reply:
We have other tests outside of resource routes that fail